src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala

```diff
@@ -111,53 +108,106 @@ private[spark] class FlexibleDataModelCorePropertySyncRelation(
+        TableExpression(edges = Some(EdgeTableExpression(filter = Some(filters))))
+      case InstanceType.Node =>
+        TableExpression(nodes = Some(NodesTableExpression(filter = Some(filters))))
+    }
 
-    Seq(syncOut(instanceSyncRequest, selectedColumns))
+  private def fetchData(
+      isBackFill: Boolean,
+      cursors: Option[Map[String, String]],
+      `with`: Map[String, TableExpression],
+      select: Map[String, SelectExpression]): IO[ItemsWithCursor[InstanceDefinition]] = {
+    val response = if (isBackFill) {
+      client.instances.queryRequest(
+        InstanceQueryRequest(
+          `with` = `with`,
+          cursors = cursors,
+          select = select,
+          includeTyping = Some(true)
+        ))
+    } else {
+      client.instances.syncRequest(
+        InstanceSyncRequest(
+          `with` = `with`,
+          cursors = cursors,
+          select = select,
+          includeTyping = Some(true)
+        ))
+    }
+    response.map { qr =>
+      val itemDefinitions = qr.items.getOrElse(Map.empty).get("sync").getOrElse(Vector.empty)
+      val nextCursor = qr.nextCursor.getOrElse(Map.empty).get("sync")
+      ItemsWithCursor(itemDefinitions, nextCursor)
+    }
   }
   // scalastyle:on cyclomatic.complexity
 
-  private def syncOut(syncRequest: InstanceSyncRequest, selectedProps: Array[String])(
+  private def syncOut(
+      futureItems: Option[String],
+      isBackFill: Boolean,
+      cursors: Option[Map[String, String]],
+      `with`: Map[String, TableExpression],
+      select: Map[String, SelectExpression],
+      selectedProps: Array[String])(
       implicit F: Async[IO]): Stream[IO, ProjectedFlexibleDataModelInstance] =
     Stream.eval {
-      syncWithCursor(syncRequest).map { sr =>
+      fetchData(isBackFill, cursors, `with`, select).map { sr =>
         val items = sr.items
         val nextCursor = sr.nextCursor
         val next =
           (nextCursor, items.nonEmpty && items.last.lastUpdatedTime < terminationTimeStamp) match {
             case (Some(cursor), true) =>
-              syncOut(syncRequest.copy(cursors = Some(Map("sync" -> cursor))), selectedProps)
+              syncOut(
+                futureItems,
+                isBackFill,
+                Some(Map("sync" -> cursor)),
+                `with`,
+                select,
+                selectedProps)
             case _ => fs2.Stream.empty
           }
-
-        val projected = items.map(toProjectedInstance(_, nextCursor, selectedProps))
+        val projectedCursorId = if (futureItems.isEmpty) nextCursor else futureItems
+        val projected = items.map(toProjectedInstance(_, projectedCursorId, selectedProps))
         fs2.Stream.emits(projected) ++ next
       }
     }.flatten
 
-  private def syncWithCursor(
-      syncRequest: InstanceSyncRequest): IO[ItemsWithCursor[InstanceDefinition]] = {
-    // temporary handle expired cursor by syncing from scratch
-    val response = client.instances
```

Comment:
I kind of liked this behavor, since it would never make the job fail.

Reply:
added back support for expired cursors handling